### PR TITLE
keychain+lnwallet: when fetching priv keys or signing try to use cache 

### LINKED ISSUE #5125: Potential private key derivation optimization
I've been running load tests for receiving payments and generated flame graphs to get a feel for where time is spent. One thing that stood out is how expensive `BtcWalletKeyRing.DerivPrivKey` is. That method not only derives a key, but also accesses the database for update.

To find out if the derived keys are all different or the same, I patched `lnd` to add an extra marker-method to the call tree `deriveRootPrivKey`. Execution flows through that marker-method only when the root of a key family is requested.

Below is the resulting flame graph where `deriveRootPrivKey` is marked in purple.

flamereceive

The same root key is requested over and over again for signing the invoice and ECDH onion operations. This seems like a candidate for optimization, for example through caching.

Another observation is that invoice add and update are quite costly. It is unfortunate that bbolt only offers full database locks which cause concurrent payments to block each other.

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we start to optimistically use the new private key cache
that was added to btcwallet. As is, btcwallet will cache the decrypted
account keys for each scope in memory. However, the existing methods
to derive a child key from those account keys requires a write database
transaction, and will re-derive the private key using BIP-32 each time.

The newly added `DeriveFromKeyPathCache` lets us skip all this and
directly use a cache assuming the account info is already cached. The
new logic will try to use this method, but if it fails fall back to the
existing `DeriveFromKeyPath` method. All calls after this will use this
new cached key.

Fixes #5125.

Basic benchmark before the btcwallet change and after:
```
benchmark                    old ns/op     new ns/op     delta
BenchmarkDerivePrivKey-8     22840583      125           -100.00%

benchmark                    old allocs     new allocs     delta
BenchmarkDerivePrivKey-8     89             2              -97.75%

benchmark                    old bytes     new bytes     delta
BenchmarkDerivePrivKey-8     10225         24            -99.77%

### CODE REVIEW

File: keychain/btcwallet.go
Comment (guggero) [inline keychain/btcwallet.go:30]: nit: unused.

File: keychain/bench_test.go
Comment (guggero) [inline keychain/bench_test.go:14]: nit: use require?

File: keychain/bench_test.go
Comment (guggero) [inline keychain/bench_test.go:39]: Move those inside the loop?

**Review by guggero — APPROVED** [review]
Niiiice :sunglasses: This should give us quite the performance boost.

LGTM after dependent PR is merged.

File: keychain/btcwallet.go
Comment (carlaKC) [inline keychain/btcwallet.go:276]: If `err` can be anything other that `ErrAccountNotCached`, do we want to log/handle it? Likewise below.

**Review by carlaKC — APPROVED** [review]
Nice optimization, LTGM!

**Pull request merged**
